test/spec/app_spec.rb

```diff
@@ -632,63 +619,58 @@
 
     it "should not add a duplicated tag from command line", bin3: true do
       media_pg = api_create_team_project_and_claim_and_redirect_to_media_page
-
       new_tag = Time.now.to_i.to_s
+      old = @driver.find_elements(:class,"annotations__list-item").length
 
       # Validate assumption that tag does not exist
       expect(media_pg.has_tag?(new_tag)).to be(false)
 
       # Try to add from command line
       media_pg.add_annotation("/tag #{new_tag}")
-      Selenium::WebDriver::Wait.new(timeout: 10).until { media_pg.has_tag?(new_tag) } # TODO: wait inside MediaPage
+      old = wait_for_size_change(old, "annotations__list-item", :class)
+      media_pg.has_tag?(new_tag)
       expect(media_pg.has_tag?(new_tag)).to be(true)
 
       # Try to add duplicate from command line
       media_pg.add_annotation("/tag #{new_tag}")
 
       # Verify that tag is not added and that error message is displayed
       expect(media_pg.tags.count(new_tag)).to be(1)
+      sleep 3
       expect(@driver.page_source.include?('Tag already exists')).to be(true)
     end
 
-    it "should not create duplicated media", bin1: true do
+    it "should not create duplicated media", bin4: true do
       api_create_team_project_and_link_and_redirect_to_media_page @media_url
       id1 = @driver.current_url.to_s.gsub(/^.*\/media\//, '').to_i
       expect(id1 > 0).to be(true)
       @driver.navigate.to @driver.current_url.to_s.gsub(/\/media\/[0-9]+$/, '')
-
       sleep 3
+      wait_for_selector("medias__item",:class)
+      media_url = @driver.current_url.to_s.gsub(/\/media\/[0-9]+$/, '')
       fill_field('#create-media-input', @media_url)
       sleep 2
       press_button('#create-media-submit')
-      sleep 10
-
-      id2 = @driver.current_url.to_s.gsub(/^.*\/media\//, '').to_i
-      expect(id2 > 0).to be(true)
-
-      expect(id1 == id2).to be(true)
+      sleep 3
+      wait_for_selector("add-annotation__insert-photo",:class)
```

Comment:
Same question as above: Is it really needed to have a `sleep` here since it's followed by a `wait_for_selector`?

Reply:
No. will fix it.